python/fastquant/strategies/base.py

```diff
@@ -172,261 +190,307 @@ def notify_order(self, order):
-                            self.log("Stop price: {}".format(stop_price))
-                        self.sell(
-                            exectype=bt.Order.Stop, price=stop_price, size=final_size,
-                        )
-
-                    if self.stop_trail:
-                        if self.transaction_logging:
-                            self.log("Stop trail: {}".format(self.stop_trail))
-                        self.sell(
-                            exectype=bt.Order.StopTrail,
-                            trailpercent=self.stop_trail,
-                            size=final_size,
-                        )
-
-        elif self.sell_signal():
-            if self.allow_short:
-
-                # Sell short based on the closing price of the previous day
-                if self.execution_type == "close":
-
-                    sell_prop_size = int(
-                        SELL_PROP * self.broker.getvalue() / self.dataclose[1]
-                    )
-                    # The max incremental short allowed is the short that would lead to a cumulative short position
-                    # equal to the maximum short position (initial cash times the maximum short ratio, which is 1.5 by default)
-                    max_position_size = max(
-                        int(self.broker.getvalue() * self.short_max / self.dataclose[1])
-                        + self.position.size,
-                        0,
-                    )
-                    if max_position_size > 0:
-                        if self.transaction_logging:
-                            self.log("SELL CREATE, %.2f" % self.dataclose[1])
-                        self.order = self.sell(
-                            size=min(sell_prop_size, max_position_size)
-                        )
-
-                # Buy based on the opening price of the next closing day (only works "open" data exists in the dataset)
-                else:
-
-                    sell_prop_size = int(
-                        SELL_PROP * self.broker.getvalue() / self.dataopen[1]
-                    )
-                    # The max incremental short allowed is the short that would lead to a cumulative short position
-                    # equal to the maximum short position (initial cash times the maximum short ratio, which is 1.5 by default)
-                    max_position_size = max(
-                        int(self.broker.getvalue() * self.short_max / self.dataopen[1])
-                        + self.position.size,
-                        0,
-                    )
-                    if max_position_size > 0:
-                        if self.transaction_logging:
-                            self.log("SELL CREATE, %.2f" % self.dataopen[1])
-                        self.order = self.sell(
-                            size=min(sell_prop_size, max_position_size)
-                        )
 
-            elif stock_value > 0:
+            # Buy on tomorrow's closing price
```

Comment:
@jbdelmundo  Yeah, I think best to still include the `open` type of execution. Technically, next day open is the most recommended approach to implementing the buy and sell; although, we use current / next day close as a proxy when next day open is not available.